apps/api/src/domain/models/vocabulary-list/entities/vocabulary-list-variable.entity.ts

```diff
@@ -1,18 +1,32 @@
 import { IVocabularyListVariable } from '@coscrad/api-interfaces';
 import { NonEmptyString } from '@coscrad/data-types';
+import { isBoolean, isNonEmptyString } from '@coscrad/validation-constraints';
 import { ApiProperty } from '@nestjs/swagger';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import cloneToPlainObject from '../../../../lib/utilities/cloneToPlainObject';
+import { DTO } from '../../../../types/DTO';
+import BaseDomainModel from '../../BaseDomainModel';
 import { DropboxOrCheckbox } from '../../vocabulary-list/types/dropbox-or-checkbox';
 import { VocabularyListVariableValue } from '../../vocabulary-list/types/vocabulary-list-variable-value';
+import { DuplicateValueForVocabularyListFilterPropertyValueError } from '../errors';
+import { CheckboxMustHaveExactlyTwoAllowedValuesError } from '../errors/checkbox-must-have-exactly-two-allowed-values.error';
+import { DuplicateLabelForVocabularyListFilterPropertyValueError } from '../errors/duplicate-label-for-vocabulary-list-filter-property-value.error';
+import { InvalidValueForCheckboxFilterPropertyError } from './invalid-value-for-checkbox-filter-property.error';
+import { InvalidValueForSelectFilterPropertyError } from './invalid-value-for-select-filter-property.error';
 import { ValueAndDisplay } from './value-and-display.entity';
 
 /**
  * TODO correlate `type` and `validValues`
  * type:dropbox <-> string values
  * type:checkbox <-> boolean values
  */
-export class VocabularyListVariable<
-    TVariableType extends VocabularyListVariableValue = VocabularyListVariableValue
-> implements IVocabularyListVariable<TVariableType>
+// TODO Rename this `VocabularyListFilterProperty` or just `FilterProperty`
```

Comment:
[Here](https://www.pivotaltracker.com/story/show/186469235) is the story for that.